src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +490,107 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             return actualArguments.AsImmutableOrNull();
         }
 
+        internal static ImmutableArray<BoundExpression> MakeArgumentsInEvaluationOrder(
```

Comment:
>ImmutableArray<BoundExpression> [](start = 24, length = 31)

It looks like we can simply return an ```ImmutableArray<IArgument>``` and get rid of argumentKindBuilder and parameterSymbolBuilder parameters.